src/pages/patientView/trialMatch/TrialMatchTable.tsx

```diff
@@ -0,0 +1,343 @@
+                ))}
+            </div>
+        );
+    }
+    public tooltipClinicalContent(data: Array<string>) {
+        return (
+            <div style={{ maxHeight:400, maxWidth:600, overflow:'auto' }}>
+                <ul className={styles.alterationUl}>
+                    {data.map((cancerType: string) => (
+                        <li>{cancerType}</li>
+                    ))}
+                </ul>
+            </div>
+        );
+    }
+
+    public mainContent(content: string) {
+        return (
+            <span>
+                <a>{content}</a>
+            </span>
+        );
+    }
+
+    public getHugoSymbolName(matches: Array<IGenomicGroupMatch>, threshold: number) {
+        const hugoSymbolSet = new Set([...matches].map(x => x.genomicAlteration.split(' ')[0].slice(1)));
+        if (hugoSymbolSet.size <= threshold) {
+            return [...hugoSymbolSet].join(', ');
+        }
+        return hugoSymbolSet.size + ' genes' ;
+    }
+
+    public buildDetailedTrialMatches() {
+        const props = this.props;
+        const groupByTrialId = _.groupBy(props.trialMatches, 'nctId');
+        let matchedTrials: Array<IDetailedTrialMatch> = [];
+        const hiddenArmTypes = ['Control Arm', 'Placebo Arm'];
+        _.forEach(groupByTrialId, function (trialGroup, trialId) {
+            let matchedTrial:any = _.find( props.trials, { 'nctId': trialId } );
+            matchedTrial['priority'] = 0; // highest priority
+            matchedTrial['matches'] = [];
+            const groupByArm = _.groupBy(trialGroup, 'armDescription');
+            _.forEach(groupByArm, function(armGroup, armDescription) {
+                let armMatch: IArmMatch = {
+                    armDescription: armDescription,
+                    drugs: [],
+                    matches: []
+                };
+                if ( armDescription !== 'null' ) { // match for specific arm
+                    const matchedArm = _.find( matchedTrial['treatmentList']['step'][0]['arm'], { 'arm_description': armDescription } );
+                    if ( hiddenArmTypes.indexOf(matchedArm['arm_type']) < 0 ) {
+                        if ( matchedArm[ 'drugs' ] ) {
+                            armMatch.drugs.push( _.map( matchedArm[ 'drugs' ], 'name' ) );
+                        }
+                    } else {
+                        return; // skip the hidden arm
+                    }
+                }
+                const groupByAge = _.groupBy(armGroup, 'trialAgeNumerical');
+                _.forEach(groupByAge, function(ageGroup, age) {
+                    const cancerTypes =  _.uniq(_.map(ageGroup, 'trialOncotreePrimaryDiagnosis'));
+                    let generalCancerTypes: Array<string> = [];
+                    let notCancerTypes: Array<string> = [];
+                    _.map(cancerTypes, function(item) {
+                       if (item.indexOf('!') !== -1) {
+                           notCancerTypes.push(item.replace(/!/g, ''));
```

Comment:
Do we ever have multiple `!` in the string?

Reply:
For cancerTypes, there is only one `!`. I'll remove `g` in RegExp.